fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanUtilService.java

```diff
@@ -325,4 +332,35 @@ private LocalDate generateCalculatedRepaymentStartDate(final CalendarHistoryData
         return disbursementDatas;
     }
 
+    public List<Rate> fetchRateData(final JsonArray element) {
+        List<Rate> rates = new ArrayList<>();
+
+        if (element != null) {
+            for (JsonElement jsonElement : element) {
+                final JsonObject rateElement = jsonElement.getAsJsonObject();
+                if (rateElement.has("id")) {
+                    final Long rateId = this.fromApiJsonHelper.extractLongNamed("id", rateElement);
+                    Rate rate = findRateByIdIfProvided(rateId);
+                    rates.add(rate);
+                }
+            }
+        }else{
+            System.out.println("");
+        }
+
+        return rates;
+    }
+
+    public Rate findRateByIdIfProvided(final Long rateId) {
```

Comment:
kindly provide appropriate name

Reply:
Ok. The name is based on the names used by the methods in the class LoanAssembler. We were just following what we assumed it was a name pattern.